### PR TITLE
Add batch API

### PR DESCRIPTION
**Pull request by coryfklein**
Includes Job and JobList, as well as accompanying Json formatters.

<3

### CODE REVIEW

**Comment by farmdawgnation** [comment]
@doriordan I'm interested in this PR as well. Is there an ETA on when you'll have an opportunity to land it?

**Comment by doriordan** [comment]
@coryfklein thanks for the PR @farmdawgnation I hope to land this in the next couple of days

File: client/src/main/scala/skuber/batch/package.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  implicit val jobKind: ObjKind[Job] = new ObjKind[Job]("jobs","Job") with IsBatchKind[Job]
+
+  // support for the corresponding List kinds
+  case class JobList( val kind: String = "JobList",
+                      override val apiVersion: String = batchAPIVersion,
+                      val metadata: Option[ListMeta] = None,
+                      items: List[Job] = Nil) extends KList[Job]
+  implicit val deplListKind = new ListKind[JobList]("deployments")
```
Comment (doriordan) [inline client/src/main/scala/skuber/batch/package.scala]: I assume this should be something like `implicit val jobListKind = new ListKind[JobList]("jobs") with IsBatchKind[JobList]`?

**Comment by coryfklein** [comment]
Ah yes, that is correct. Copy pasta error there. I'll fix that in a bit
when I get in the office.

**Comment by doriordan** [comment]
Thanks - otherwise looks good.

**Comment by coryfklein** [comment]
Okay, had to force push to resolve some conflicts against master. But should be all good to go now. I fixed the deplListKind issue.

**Pull request merged**